design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,280 @@
+specifically subscription-id, resource-group, and user/system-assigned-identity. These fields will be used to confirm the identity of the Azure resource 
+sending the authentication request by comparing the Azure token information with the host annotations that are defined in the above policy.
+
+A Conjur host will be defined as follows with their annotations holding Azure-specific identification properties.
+
+```yaml
+- !policy 
+  id: azure-apps 
+  body: 
+  - !host 
+    id: test-app 
+    annotations: 
+    authn-azure/subscription-id: test-subscription 
+    authn-azure/resource-group: test-group 
+    authn-azure/user-assigned-identity: test-app-pipeline 
+OR 
+    authn-azure/system-assigned-identity: ie_qWCXhXxt1zIEsu4c7acQVGn4
+```
+ 
+#### Multiple-assigned identity for host - NOT FOR NOW
+
+We may allow hosts to be defined with both "authn-azure/user-assigned-identity" and "authn-azure/system-assigned-identity". This way such a host can 
+authenticate a token with a user-assigned id as well as a token with a system-assigned id. As a first step we will not develop - nor test - such a use-case.
+
+### Design
+
+#### Authentication Flow
+
+This flow assumes that the VM already exists in Azure, and that an authn-azure authenticator and a Conjur host were loaded.
+
+![Authn Azure Flow](authn-azure-flow.png)
+
+
+1. Azure VM sends the following request to Conjur:
+    ```yaml
+    POST https://<DAP-server-hostname>/authn-azure/<service-id>/<account>/<host-id>authenticate
+    Header 
+    
+    Content-Type: application/x-www-form-urlencoded
+    Body 
+    
+     token: "eyJhbGciOiJSUzI1NiIs......uTonCA"
+    ```
+
+1. An already existing route will be used (post '/:authenticator(/:service_id)/:account/:id/authenticate' => 'authenticate#authenticate') that will forward the request to the AuthenticationController. 
+
+1. The request will then be forwarded to the Authentication::Authenticate CommandClass. This CommandClass holds a collection of general authenticator validations that will take place.
+
+1. The two first general validations include validate_authenticator_exists and validate_security. 
+    
+    1. validate_authenticator_exists ensures that the authenticator sent as part of the request URI is one that we support (authn-k8s, authn-azure, etc.).
+
+    1. validate_security validates that the Azure authenticator has been defined correctly in policy and the host is permitted to authenticate to the webservice.
+
+1. The next validation, validate_credentials calls the authenticator's valid? function. If so, Azure-specific authenticator validations will be triggered. 
+
+    1. validate_azure_token (see below for a more detailed explanation of the function).
+    
+    1. validate_application_identity, the Conjur identity host annotations for the Azure VM and the properties extracted from the "xms_mirid" field of the Azure token provided by the Azure VM are compared (see below for a more detailed explanation of the function).
```

Comment:
It would be great to add "xms_mirid" to the Glossary and explain what it is and why it's Ok we use it.

Reply:
Good point- Adding this!